### DISCUSSION TITLE
Custom Criterion

### DISCUSSION THREAD

**Discussion by nina-wiegers** [ContinualAI/avalanche#1113] [Q&A] [answered]
Hey everyone, 

I want to change the criterion for every experience. I used the code from https://github.com/ContinualAI/avalanche/pull/604#issue-886775416 and changed it a bit.  I need different weights for each experience, because I have got a dataset with unbalanced data.  In the below, i posted my code. My question is, how can I validate, that this code works. When I print out strategy.criterion in the train loop, I cannot verify, that these loss functions are in use. Has anybody got any idea or had this problem before?

  loss_fun_0 = CrossEntropyLoss(weight=exp_0_weights)
  loss_fun_1 = CrossEntropyLoss(weight=all_weights)

  class ClassWeights(SupervisedPlugin):
  
      def __init__(self, loss_fun_0, loss_fun_1):
          super().__init__()
          self.loss_fun_0 = loss_fun_0
          self.loss_fun_1 = loss_fun_1
  
      def criterion(self):
          if self.experience == 0:
              return self.loss_fun_0(self.mb_pred, self.mb_y)
          else:
              return self.loss_fun_1(self.mb_pred, self.mb_y)

**Comment by AntonioCarta** [comment, marked as answer]
You can use a debugger and put a breakpoint in your function. Alternatively, you can put a print inside your function to check if it is called.
